cmd/scale.go

```diff
@@ -0,0 +1,271 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"os"
+	"path"
+	"time"
+
+	"github.com/Azure/acs-engine/pkg/acsengine"
+	"github.com/Azure/acs-engine/pkg/api"
+	"github.com/Azure/acs-engine/pkg/armhelpers"
+	"github.com/Azure/acs-engine/pkg/i18n"
+	"github.com/leonelquinteros/gotext"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type scaleCmd struct {
+	authArgs
+
+	// user input
+	resourceGroupName    string
+	deploymentDirectory  string
+	newDesiredAgentCount int
+	containerService     *api.ContainerService
+	apiVersion           string
+	location             string
+	agentPoolToScale     string
+	classicMode          bool
+
+	// derived
+	apiModelPath   string
+	agentPool      *api.AgentPoolProfile
+	client         armhelpers.ACSEngineClient
+	locale         *gotext.Locale
+	nameSuffix     string
+	agentPoolIndex int
+}
+
+const (
+	scaleName             = "scale"
+	scaleShortDescription = "scale a deployed cluster"
+	scaleLongDescription  = "scale a deployed cluster"
+)
+
+// NewScaleCmd run a command to upgrade a Kubernetes cluster
+func newScaleCmd() *cobra.Command {
+	sc := scaleCmd{}
+
+	scaleCmd := &cobra.Command{
+		Use:   scaleName,
+		Short: scaleShortDescription,
+		Long:  scaleLongDescription,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return sc.run(cmd, args)
+		},
+	}
+
+	f := scaleCmd.Flags()
+	f.StringVar(&sc.location, "location", "", "location the cluster is deployed in")
+	f.StringVar(&sc.resourceGroupName, "resource-group", "", "the resource group where the cluster is deployed")
+	f.StringVar(&sc.deploymentDirectory, "deployment-dir", "", "the location of the output from `generate`")
+	f.IntVar(&sc.newDesiredAgentCount, "new-agent-count", 0, "desired number of agents")
```

Comment:
Looks like it is referenced as `new-agent-count` here, but validation of the argument prints a message calling it `agent-count`

Reply:
thanks I fixed it